src/renderer/components/+preferences/app-preferences/app-preference-registration.d.ts

```diff
@@ -33,16 +32,3 @@ export interface AppPreferenceRegistration {
   showInPreferencesTab?: string;
   components: AppPreferenceComponents;
 }
-
-export interface RegisteredAppPreference extends AppPreferenceRegistration {
-  id: string;
-}
```

Comment:
I don't think we should remove this type. I think that the injectable's type should be `IComputedValue<RegisteredAppPreference[]>`.

That way we communicate that `id` is always defined (which is good when we (hopefully soon) turn on `strict`)

Reply:
Thanks, I wasn't sure the best way to handle this type. Internally it seems we always use `RegisteredAppPreference`